File: Sources/XMTPiOS/Libxmtp/ConversationDebugInfo.swift
import LibXMTP

public enum CommitLogForkStatus {
	case forked
	case notForked
	case unknown
}

public struct ConversationDebugInfo {
	let ffiConversationDebugInfo: FfiConversationDebugInfo
	
	public init(ffiConversationDebugInfo: FfiConversationDebugInfo) {
		self.ffiConversationDebugInfo = ffiConversationDebugInfo
	}

	public var epoch: UInt64 {
		ffiConversationDebugInfo.epoch
	}

	public var maybeForked: Bool {
		ffiConversationDebugInfo.maybeForked
	}

	public var forkDetails: String {
		ffiConversationDebugInfo.forkDetails
	}
    
    public var localCommitLog: String {
        ffiConversationDebugInfo.localCommitLog
    }
    
    public var remoteCommitLog: String {
        ffiConversationDebugInfo.remoteCommitLog
    }
	
	public var commitLogForkStatus: CommitLogForkStatus {
		switch ffiConversationDebugInfo.isCommitLogForked {
		case true:
			return .forked
		case false:
			return .notForked
		default:
			return .unknown
		}
	}
}

File: sdks/android/library/src/main/java/org/xmtp/android/library/libxmtp/ConversationDebugInfo.kt
package org.xmtp.android.library.libxmtp

import uniffi.xmtpv3.FfiConversationDebugInfo

class ConversationDebugInfo(
    private val ffiConversationDebugInfo: FfiConversationDebugInfo,
) {
    enum class CommitLogForkStatus {
        FORKED,
        NOT_FORKED,
        UNKNOWN,
    }

    val epoch: Long
        get() = ffiConversationDebugInfo.epoch.toLong()
    val maybeForked: Boolean
        get() = ffiConversationDebugInfo.maybeForked
    val forkDetails: String
        get() = ffiConversationDebugInfo.forkDetails
    val localCommitLog: String
        get() = ffiConversationDebugInfo.localCommitLog
    val remoteCommitLog: String
        get() = ffiConversationDebugInfo.remoteCommitLog
    val commitLogForkStatus: CommitLogForkStatus
        get() =
            when (ffiConversationDebugInfo.isCommitLogForked) {
                true -> CommitLogForkStatus.FORKED
                false -> CommitLogForkStatus.NOT_FORKED
                null -> CommitLogForkStatus.UNKNOWN
            }
}
